client/src/pages/Configurations/index.test.tsx

```diff
@@ -10,24 +18,116 @@ const renderPageView = () =>
     renderPageView();
     expect(
-      screen.getByText('Set up new condition', {
-        selector: 'button.usa-button[type=button]',
-      })
+      screen.getByText('Your reportable condition configurations')
     ).toBeInTheDocument();
-  });
-  it('should have a search component with correct placeholder', async () => {
-    renderPageView();
     expect(
       screen.getByPlaceholderText('Search configurations')
     ).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Set up new condition' })
+    ).toBeInTheDocument();
+  });
+
+  test('opens the modal when "Set up new condition" button is clicked', async () => {
+    const user = userEvent.setup();
+    renderPageView();
+    const setUpButton = screen.getByRole('button', {
+      name: 'Set up new condition',
+    });
+    await user.click(setUpButton);
+    const modal = screen.getByRole('dialog');
+    expect(modal).toBeInTheDocument();
+    expect(getByText(modal, 'Set up new condition')).toBeInTheDocument();
+  });
+
+  test('selects a condition from the ComboBox and enables the Add condition button', async () => {
+    const user = userEvent.setup();
+    renderPageView();
+    const setUpButton = screen.getByRole('button', {
+      name: 'Set up new condition',
+    });
+    await user.click(setUpButton);
+
+    expect(await screen.findByRole('dialog')).toBeInTheDocument();
+
+    const conditionInput = screen.getByLabelText('Condition');
+    expect(conditionInput).toBeInTheDocument();
+
+    await user.type(conditionInput, 'Anaplasmosis');
+
+    const anaplasmosisOption = await findByText(
+      await screen.findByRole('option'),
+      'Anaplasmosis'
+    );
+    expect(anaplasmosisOption).toBeInTheDocument();
+
+    await userEvent.click(anaplasmosisOption);
+
+    expect(conditionInput).toHaveValue('Anaplasmosis');
+
+    const addConditionButton = screen.getByRole('button', {
+      name: 'Add condition',
+    });
+    expect(addConditionButton).toBeEnabled();
+  });
+
+  test.skip('submits the form and adds a new configuration to the table', async () => {
+    const user = userEvent.setup();
+    renderPageView();
+    const setUpButton = screen.getByRole('button', {
+      name: 'Set up new condition',
+    });
+    await user.click(setUpButton);
+
+    expect(await screen.findByRole('dialog')).toBeInTheDocument();
+
+    const conditionInput = screen.getByLabelText('Condition');
+    expect(conditionInput).toBeInTheDocument();
+
+    await user.type(conditionInput, 'Anaplasmosis');
+
+    const anaplasmosisOption = await findByText(
+      await screen.findByRole('option'),
+      'Anaplasmosis'
+    );
+    expect(anaplasmosisOption).toBeInTheDocument();
+
+    await userEvent.click(anaplasmosisOption);
+
+    expect(conditionInput).toHaveValue('Anaplasmosis');
+
+    const addConditionButton = await screen.getByRole('button', {
+      name: 'Add condition',
+    });
```

Comment:
`getByRole` is sync so no need to `await`!
```suggestion
    const addConditionButton = screen.getByRole('button', {
      name: 'Add condition',
    });
```